.github/workflows/release-helm.yml

```diff
@@ -0,0 +1,116 @@
+---
+name: Release Helm Chart
+
+on:
+  workflow_dispatch: {}
+  push:
+    branches:
+      - main
+    paths:
+        - 'deploy/helm/Chart.yaml'
```

Comment:
Do we want the workflow to be pinned to releases(IE, tags) similarly to what we do for the docker container?

Reply:
Though, I guess the intent here is to only run this on man with the Chart.yaml file changes. Is that somewhat coupled to the image promotion?

Reply:
Yeah, we'll need to update the `Chart.yaml` with the new release manually when we create a tag, which will then release the Helm Chart!

Reply:
I went back and forth with the Helm release cadence. Some other projects trigger the Helm chart's release GHA with any change to `helm/*`. I limited it to `Chart.yaml` so that we only release a new chart when `Chart.yaml`'s `appVersion` (a new release) has been created, that way we keep version parity. What do you think?

Reply:
Update: I don't think that we need to (or should!) pin the Helm chart version and image version to each other. Here are other examples of `Chart.yaml` from other Grafana Labs-owned charts where these are **not coupled**:
* https://github.com/grafana/helm-charts/blob/main/charts/agent-operator/Chart.yaml
* https://github.com/grafana/loki/blob/main/production/helm/loki/Chart.yaml
* https://github.com/grafana/helm-charts/blob/main/charts/snyk-exporter/Chart.yaml
* https://github.com/grafana/helm-charts/blob/main/charts/grafana/Chart.yaml

I'll update things accordingly!